src/index.js

```diff
@@ -19,6 +20,45 @@ const client = new Client();
       console.log('>> ', message.content);
     });
 
+    // The code below listens for reactions to any message in the server and if
+    // a reaction is equal to the specified trigger reaction (in this case '🤖'),
+    // then is attempts to format the message. To get a detailed explanation of the
+    // code below visit: https://discordjs.guide/popular-topics/reactions.html#listening-for-reactions-on-old-messages
+
+    client.on('messageReactionAdd', async (reaction) => {
+      if (reaction.partial) {
+        try {
+          await reaction.fetch();
+        } catch (error) {
+          try {
+            reaction.message.channel.send(
+              'Something went wrong! Failed to format code :('
+            );
+          } catch (error) {
+            // maybe write an error log or something
+          }
```

Comment:
I don't think we the nested `try/catch`, just `console.error` the `error` return from the fetch call and send a message back to chat if possible.

If the bot can't send error messages to chat, then its fine if it explodes, as long as the system/service managing the bot notifies someone. For the official FCC version of the bot we probably wouldn't want to spam chat with error messages, but this version can. If we supress error messages via this nested try and catch **no one** will know what sort of errors are happening or why.

Assuming we host the bot on some Paas, or FCC uses some system like Docker/kubernetes, then the "host" system will manage critical errors like this, rather then the bot itself.

Reply:
Done...